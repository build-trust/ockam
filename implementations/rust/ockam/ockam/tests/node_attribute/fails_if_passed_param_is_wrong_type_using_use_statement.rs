// This test checks that #[ockam::node] causes a compile time error
// if the function is passed a param that is not of type `ockam::Context`
// The param is not fully qualified (ie. using `use` statement).

use std::string::String;

#[ockam::node]
async fn main(ctx: String) {}
