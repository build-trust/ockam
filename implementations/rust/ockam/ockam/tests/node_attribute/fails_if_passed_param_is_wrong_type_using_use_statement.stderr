error: Expected `ockam::Context` found `String`
 --> $DIR/fails_if_passed_param_is_wrong_type_using_use_statement.rs:8:20
  |
8 | async fn main(ctx: String) {}
  |                    ^^^^^^

warning: unused import: `std::string::String`
 --> $DIR/fails_if_passed_param_is_wrong_type_using_use_statement.rs:5:5
  |
5 | use std::string::String;
  |     ^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fails_if_passed_param_is_wrong_type_using_use_statement.rs:5:1
  |
5 | / use std::string::String;
6 | |
7 | | #[ockam::node]
8 | | async fn main(ctx: String) {}
  | |_____________________________^ consider adding a `main` function to `$DIR/tests/node_attribute/fails_if_passed_param_is_wrong_type_using_use_statement.rs`
